cassandra-analytics-core/src/main/java/org/apache/cassandra/clients/Sidecar.java

```diff
@@ -129,16 +130,15 @@ public static SidecarClient from(SidecarInstancesProvider sidecarInstancesProvid
                                             .ssl(conf.hasKeystoreAndKeystorePassword())
                                             .build();
 
-        SidecarConfig sidecarConfig = new SidecarConfig.Builder()
-                                      .maxRetries(5)
-                                      .retryDelayMillis(200)
-                                      .maxRetryDelayMillis(500)
-                                      .build();
-
-        return buildClient(sidecarConfig, vertx, httpClientConfig, sidecarInstancesProvider);
+        SidecarClientConfig sidecarConfig = SidecarClientConfigImpl.builder()
+                                                                       .maxRetries(5)
+                                                                       .retryDelayMillis(200)
+                                                                       .maxRetryDelayMillis(500)
+                                                                       .build();
+        return buildClient((SidecarClientConfigImpl) sidecarConfig, vertx, httpClientConfig, sidecarInstancesProvider);
     }
 
-    public static SidecarClient buildClient(SidecarConfig sidecarConfig,
+    public static SidecarClient buildClient(SidecarClientConfigImpl sidecarConfig,
```

Comment:
Can we use the interface here instead?
```suggestion
    public static SidecarClient buildClient(SidecarClientConfig sidecarConfig,
```

Reply:
👍